### PR TITLE
ping_interval: use service.restart instead of signaling

### PR DESCRIPTION
**Pull request by terminalmage**
When multiprocessing does not relay the signal properly, ping_interval will fail to respawn the multiprocessing.Process instance and the minion will end up in a bad state, where there are still salt-minion procs running but the minion is not connected to the master and thus does not respond to pubs from it.

This commit ditches signaling in favor of using `service.restart` to restart the minion daemon.

### CODE REVIEW

**Review by isbm — CHANGES_REQUESTED** [review]
@terminalmage it is a _benevolent_ request. :wink:

**Review by rallytime — COMMENTED** [review]
I think this looks good, but it's probably a good idea to move that to a helper function for the future.

**Comment by terminalmage** [comment]
The logic to get the service name has been moved to a separate function in `salt/minion.py`. I decided to make it a public function rather than private, since it could be useful outside of minion.py.

File: salt/minion.py [hunk 1]
```diff
@@ -2619,8 +2619,20 @@ def ping_timeout_handler(*_):
                             log.error('** Master Ping failed. Attempting to restart minion**')
                             delay = self.opts.get('random_reauth_delay', 5)
                             log.info('delaying random_reauth_delay %ss', delay)
-                            # regular sys.exit raises an exception -- which isn't sufficient in a thread
-                            os._exit(salt.defaults.exitcodes.SALT_KEEPALIVE)
+                            try:
+                                self.functions['service.restart'](
+                                    'salt_minion' if 'bsd' in sys.platform
+                                    else 'salt-minion'
+                                )
```
[the remark this answers is not in the document]
Comment (terminalmage) [reply salt/minion.py]: Good idea. I just made this change.

**Review by rallytime — APPROVED** [review]

**Review by isbm — APPROVED** [review]

**Pull request merged**